audformat/core/database.py

```diff
@@ -289,6 +290,44 @@ def drop_tables(
         for table_id in table_ids:
             self.tables.pop(table_id)
 
+    def files_duration(
+            self,
+            files: typing.Union[str, typing.Sequence[str]],
+    ) -> pd.Series:
+        r"""Duration of files in the database.
+
+        Note that durations are cached,
+        i.e. changing the files on disk after calling
+        this function can lead to wrong results.
+        The cache is cleared when the
+        database is reloaded from disk.
+
+        Use ``db.files_duration(db.files).sum()``
+        to get the total duration of all files in a database.
+        Or ``db.files_duration(db[table_id].files).sum()``
+        to get the total duration of all files assigned to a table.
+
+        Args:
+            files: file names
+
+        Returns:
+            mapping from file to duration
+
+        """
+        def duration(file: str) -> pd.Timedelta:
+            if file not in self._files_duration:
+                if not os.path.isabs(file) and self.root is not None:
+                    path = os.path.join(self.root, file)
+                else:
+                    path = file
+                dur = audiofile.duration(path)
```

Comment:
I think we should add another check here.

Let us assume that all paths are stored absolute in `db._files_duration` because the database was loaded with `full_path=True` and then we called `db.files_duration(db.files)`. If I then call `db.files_duration(relative_path_to_file)` it will generate the correct path with
```python
path = os.path.join(self.root, file)
```
but after that we should not directly call `audiofile.duration(path)` but could check again if `path` is inside `self._files_duration` and use that value instead.

Reply:
I added a check as you suggested. In addition I applied two more changes:

1. I set the name of the returned `pd.Series` to `file`
2. I added an argument `root` to cover the case where file names are relative but the database was not saved yet or loaded from disk (e.g. when a user prepares a database for publishing). In this case the user has to provide `root` or otherwise an error is raised.

Reply:
Great, thanks.